daemon/daemon_linux.go

```diff
@@ -131,3 +138,29 @@ func shouldUnmountRoot(root string, info *mount.Info) bool {
 	}
 	return hasMountinfoOption(info.Optional, sharedPropagationOption)
 }
+
+// setupResolvConf sets the appropriate resolv.conf file if not specified
+// When systemd-resolved is running the default /etc/resolv.conf points to
+// localhost. In this case fetch the alternative config file that is in a
+// different path so that containers can use it
+// In all the other cases fallback to the default one
+func setupResolvConf(config *config.Config) {
+	if config.ResolvConf != "" {
+		return
+	}
+
+	config.ResolvConf = defaultResolvConf
+	pids, err := procfs.PidOf("systemd-resolved")
+	if err != nil {
+		logrus.Errorf("unable to check systemd-resolved status: %s", err)
+		return
+	}
+	if len(pids) > 0 && pids[0] > 0 {
+		_, err := os.Stat(alternateResolvConf)
+		if err == nil {
+			logrus.Infof("systemd-resolved is running, so using resolvconf: %s", alternateResolvConf)
+			config.ResolvConf = alternateResolvConf
+		}
+		logrus.Infof("systemd-resolved is running, but %s is not present, fallback to %s", alternateResolvConf, defaultResolvConf)
```

Comment:
`else`?

Reply:
no else, line 152 (https://github.com/moby/moby/pull/37485/files/e241a0d3a74fc3b37f597464951187e132bbd7ac#diff-857979ffd7f9baf3aa83894d548f110aR152) is setting the value

Reply:
even if `alternateResolvConf` is present, "alternateResolvConf is not present" is printed here?

Reply:
sorry was looking at the wrong line